numpy/core/src/common/simd/avx2/arithmetic.h

```diff
@@ -118,6 +118,14 @@
 #endif // !NPY_HAVE_FMA3
 
 // Horizontal add: Calculates the sum of all vector elements.
+
+NPY_FINLINE npy_uint32 npyv_sum_u8(__m256i a)
+{
+    __m256i half = _mm256_sad_epu8(a, _mm256_setzero_si256());
+    __m128i quarter = _mm_add_epi32(_mm256_castsi256_si128(half), _mm256_extracti128_si256(half, 1));
+    return (unsigned)_mm_cvtsi128_si32(_mm_add_epi32(quarter, _mm_unpackhi_epi64(quarter, quarter)));
+}
```

Comment:
```suggestion
NPY_FINLINE npy_uint16 npyv_sumup_u8(npyv_u8 a)
{
    __m256i four = _mm256_sad_epu8(a, _mm256_setzero_si256());
    __m128i two  = _mm_add_epi16(_mm256_castsi256_si128(four), _mm256_extracti128_si256(four, 1));
    __m128i one  = _mm_add_epi16(two, _mm_unpackhi_epi64(two, two));
    return (npy_uint16)_mm_cvtsi128_si32(one);
}
```
increase readability